orchestration/src/main/java/com/sap/ai/sdk/orchestration/PolymorphicFallbackDeserializer.java

```diff
@@ -0,0 +1,105 @@
+package com.sap.ai.sdk.orchestration;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.google.common.annotations.Beta;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import lombok.AllArgsConstructor;
+
+/**
+ * Handles polymorphic deserialization for a base class or interface.
+ *
+ * <p>This deserializer uses a fallback strategy by attempting to deserialize the JSON into known
+ * subtypes. Subtypes are either discovered using the {@link JsonSubTypes} annotation or provided
+ * explicitly. If deserialization fails for all candidates, a {@link JsonMappingException} is thrown
+ * with suppressed exceptions.
+ *
+ * @since 1.2.0
+ * @param <T> The base type for deserialization.
+ */
+@Beta
+@AllArgsConstructor(access = lombok.AccessLevel.PRIVATE)
+public class PolymorphicFallbackDeserializer<T> extends JsonDeserializer<T> {
```

Comment:
(Question)

Will you create a follow up to move this class from `orchestration` to `common`?